File: code/eprogramoflifeapp/app/src/main/java/domain/DailyCommitment.java
package domain;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.StringTokenizer;

/**
 * Created by mmfsf on 3/29/17.
 */

public class DailyCommitment extends Commitment {

    public DailyCommitment(String name, Frequency frequency)
    {
        super(name, frequency);
        simpleDateFormat = new SimpleDateFormat("yyyyMMdd");
    }

    public Level getPerformed(Date date)
    {
        String key = simpleDateFormat.format(date);

        Level level = this.performed.get(key);
        if(level == null)
        {
            level = Level.NotDone;
        }
        return level;
    }

    public void Point(Date date, Level level)
    {
        String key = simpleDateFormat.format(date);
        this.performed.put(key, level);
    }
}

File: code/eprogramoflifeweb/epl.core/Domain/DailyCommitment.cs
using System;
using System.Collections.Generic;
using System.Text;
using epl.core.ValuesObjects;

namespace epl.core.Domain
{
  public class DailyCommitment : Commitment
  {
    protected override string KeyFormat { get => "yyyyMMdd"; }

    public DailyCommitment(string name) : base(name)
    {
      this.Frequency = Frequency.Daily;
    }

    public DailyCommitment(string name, string description) : base(name, description)
    {
      this.Frequency = Frequency.Daily;
    }

    public override Level GetPoint(DateTime date)
    {
      var key = date.Date.ToString(this.KeyFormat);
      return this.Performed.ContainsKey(key) ? this.Performed[key] : Level.NotDone;
    }

    public override void Point(DateTime date, Level level)
    {
      var key = date.Date.ToString(this.KeyFormat);
      if (this.Performed.ContainsKey(key))
      {
        this.Performed[key] = level;
      }
      else
      {
        this.Performed.Add(key, level);
      }
    }

  }
}
